modules/apps/object/object-service/src/main/java/com/liferay/object/internal/scope/DepotObjectScopeProviderImpl.java

```diff
@@ -52,6 +53,23 @@ public String[] getRootPanelCategoryKeys() {
 		return new String[0];
 	}
 
+	@Override
+	public Long getValidGroupId(long companyId, String scopeKey) {
+		Group group = _groupLocalService.fetchGroup(companyId, scopeKey);
+
+		if (group == null) {
+			group = _groupLocalService.fetchGroup(GetterUtil.getLong(scopeKey));
+		}
+
+		if (isValidGroupId(group.getGroupId()) ||
+			isValidGroupId(group.getLiveGroupId())) {
+
+			return group.getGroupId();
```

Comment:
@devpedrotavares why are you also checking the `group.getLiveGroupId` but you are returning the `group.getGroupId()`.

Reply:
I kept the same behavior we had before.
That is, we should return the group that matches the `key` passed as parameter. And if the group is a staging one, then consider it valid if the corresponding live group passes the criteria.